src/cqparts/display/cqparts_server.py

```diff
@@ -47,31 +49,59 @@ def display_callback(self, component):
         if ENVVAR_SERVER not in os.environ:
             raise KeyError("environment variable '%s' not set" % ENVVAR_SERVER)
 
-        # Verify Parameter(s)
-        # check that it is a component
-        from .. import Component
-        if not isinstance(component, Component):
-            raise TypeError("given component must be a %r, not a %r" % (
-                Component, type(component)
-            ))
+        # get the server from the environment
+        server_url = os.environ[ENVVAR_SERVER]
+        # check that the server is running
+        try:
+            requests.get(server_url + '/status')
 
-        cp_name = type(component).__name__
+            # Verify Parameter(s)
+            # check that it is a component
+            from .. import Component
+            if not isinstance(component, Component):
```

Comment:
This should be done before any communication to the server.
(causing regression tests to fail)

Reply:
OK, I set it out like this so if the server was not available it would not even try to waste cycles building a model it could not post. However, testing is more important, the user __has__ called display() anyway.

Reply:
derp it was before the export anyway..